src/plugins/discover_legacy/public/plugin.ts

```diff
@@ -318,6 +320,25 @@ export class DiscoverPlugin
         if (!this.initializeInnerAngular) {
           throw Error('Discover plugin method initializeInnerAngular is undefined');
         }
+
+        // If a user explicitly tries to access the legacy app URL
+        const {
+          core: {
+            application: { navigateToApp },
+          },
+        } = await this.initializeServices();
+        const path = window.location.hash;
+
+        const v2Enabled = core.uiSettings.get<boolean>(NEW_DISCOVER_APP);
+        if (v2Enabled) {
+          navigateToApp('data-explorer', {
+            replace: true,
+            path: `/discover`,
+            state: {
+              path,
+            } as ViewRedirectParams,
+          });
+        }
```

Comment:
This redirection already occurs in discover today. the change once we remove the legacy app is that the redirect will always happen there instead of sometimes.